sdk/python/tests/test_registry.py

```diff
@@ -0,0 +1,347 @@
+# Copyright 2019 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+from datetime import timedelta
+from tempfile import mkstemp
+
+import pytest
+from pydantic import StrictStr
+from pytest_lazyfixture import lazy_fixture
+
+from feast.data_format import ParquetFormat
+from feast.data_source import FileSource
+from feast.entity import Entity
+from feast.feature import Feature
+from feast.feature_view import FeatureView
+from feast.protos.feast.types import Value_pb2 as ValueProto
+from feast.registry import Registry
+from feast.value_type import ValueType
+
+
+@pytest.fixture
+def local_registry():
+    fd, registry_path = mkstemp()
+    return Registry(registry_path, None)
+
+
+@pytest.fixture
+def gcs_registry():
+    from google.cloud import storage
+
+    storage_client = storage.Client()
+    bucket_name = f"feast-registry-test-{int(time.time() * 1000)}"
+    bucket = storage_client.bucket(bucket_name)
+    bucket = storage_client.create_bucket(bucket)
+    bucket.add_lifecycle_delete_rule(
+        age=14
+    )  # delete buckets automatically after 14 days
+    bucket.patch()
+    bucket.blob("registry.db")
+    return Registry(f"gs://{bucket_name}/registry.db", None)
+
+
+@pytest.fixture
+def s3_registry():
+    return Registry(
+        f"s3://feast-integration-tests/registries/{int(time.time() * 1000)}/registry.db",
+        None,
+    )
+
+
+@pytest.mark.parametrize(
+    "test_registry", [lazy_fixture("local_registry")],
+)
+def test_apply_entity_success(test_registry):
+    entity = Entity(
+        name="driver_car_id",
+        description="Car driver id",
+        value_type=ValueType.STRING,
+        labels={"team": "matchmaking"},
+    )
+
+    project: StrictStr = "project"
```

Comment:
nit, `strictstr` is needed just for `pydantic` models. This can be `project = "project"`